lib/rules/no-unused-vars.js

```diff
@@ -858,10 +861,439 @@ module.exports = {
                             messageId: "unusedVar",
                             data: unusedVar.references.some(ref => ref.isWrite())
                                 ? getAssignedMessageData(unusedVar)
-                                : getDefinedMessageData(unusedVar)
+                                : getDefinedMessageData(unusedVar),
+                            suggest: [
```

Comment:
I'm not sure if we should remove the declaration if there are references to that variable.

```js
var x;

x = 1;
```

After the fix:

```js


x = 1;
```

Reply:
Done till here!